x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/action/RollupShardTask.java

```diff
@@ -15,32 +15,51 @@
 
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
 
 public class RollupShardTask extends CancellableTask {
     private final String rollupIndex;
+    private volatile long totalShardDocCount;
+    private volatile long docsProcessed;
+    private final long indexStartTimeMillis;
+    private final long indexEndTimeMillis;
     private final DownsampleConfig config;
     private final ShardId shardId;
     private final long rollupStartTime;
     private final AtomicLong numReceived = new AtomicLong(0);
     private final AtomicLong numSent = new AtomicLong(0);
     private final AtomicLong numIndexed = new AtomicLong(0);
     private final AtomicLong numFailed = new AtomicLong(0);
+    private final AtomicLong lastSourceTimestamp = new AtomicLong(0);
+    private final AtomicLong lastTargetTimestamp = new AtomicLong(0);
+    private final AtomicLong lastIndexingTimestamp = new AtomicLong(0);
+    private final AtomicReference<RollupShardIndexerStatus> rollupShardIndexerStatus = new AtomicReference<>(
+        RollupShardIndexerStatus.INITIALIZED
+    );
+    private final RollupBulkStats rollupBulkStats;
+    private final AtomicReference<RollupBeforeBulkInfo> lastBeforeBulkInfo = new AtomicReference<>(null);
+    private final AtomicReference<RollupAfterBulkInfo> lastAfterBulkInfo = new AtomicReference<>(null);
 
     public RollupShardTask(
         long id,
         String type,
         String action,
         TaskId parentTask,
         String rollupIndex,
+        long indexStartTimeMillis,
+        long indexEndTimeMillis,
         DownsampleConfig config,
         Map<String, String> headers,
         ShardId shardId
     ) {
         super(id, type, action, RollupField.NAME + "_" + rollupIndex + "[" + shardId.id() + "]", parentTask, headers);
         this.rollupIndex = rollupIndex;
+        this.indexStartTimeMillis = indexStartTimeMillis;
+        this.indexEndTimeMillis = indexEndTimeMillis;
         this.config = config;
         this.shardId = shardId;
         this.rollupStartTime = System.currentTimeMillis();
```

Comment:
Maybe use `client.threadPool().absoluteTimeInMillis()` here too?

Reply:
Here we don't have a client.

Reply:
I think this one can be converted? `client` is accessible.

Reply:
This isn't currently possible. There is no client available in the request class that creates this task.